curator-framework/src/test/java/org/apache/curator/framework/imps/TestCleanState.java

```diff
@@ -43,10 +44,8 @@ public static void closeAndTestClean(CuratorFramework client)
             EnsembleTracker ensembleTracker = internalClient.getEnsembleTracker();
             if ( ensembleTracker != null )
             {
-                while ( ensembleTracker.hasOutstanding() )
-                {
-                    Thread.sleep(100);
-                }
+                Awaitility.await()
```

Comment:
here you are adding an implicit timeout of 10 seconds
I am fine with it, as far as it does not make the test become flaky